google/oauth2/credentials.py

```diff
@@ -71,6 +71,10 @@ def __init__(self, token, refresh_token=None, id_token=None,
                 to obtain authorization. This is a purely informative parameter
                 that can be used by :meth:`has_scopes`. OAuth 2.0 credentials
                 can not request additional scopes after authorization.
+            downscope (bool): Whether to reduce the requested scopes from those
+                of the refresh token to those listed in scopes. Useful if
+                refresh token has a wild card scope (e.g.
+                'https://www.googleapis.com/auth/any-api').
```

Comment:
I'm ambivalent on the name 'downscope':  is it widely used elsewhere?  For a boolean, I prefer an obvious verb phrase, e.g., `reduce_scopes`.   @theacodes WDYT?

Reply:
I don't know about widely used, but I've also never seen the concept referred to in any other way (ex. https://developers.google.com/identity/protocols/OAuth_ref).